### PR TITLE
docs: add method documentation to exceptions module

### PR DESCRIPTION
**Pull request by Ninjagod1251**
### What I did

added exception.md to the index 

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by unparalleled-js** [comment]
I believe the diff looks weird on this one because it is not updated with main

File: src/ape/api/providers.py [hunk 1]
```diff
@@ -20,12 +20,24 @@
...
     DYNAMIC = "0x2"  # EIP-1559
 
 
 @abstractdataclass
 class TransactionAPI:
+    """
+    An API class representing a transaction.
+    Ecosystem plugins implement one or more of transaction APIs
+    depending on which schemas they permit,
+    such as typed-transactions from `EIP-1559` <https://eips.ethereum.org/EIPS/eip-1559>`__.
```
Comment (fubuloubu) [inline src/ape/api/providers.py:38]:
```suggestion
    such as `EIP-1559` <https://eips.ethereum.org/EIPS/eip-1559>`__ dynamic fee transactions.
```

File: src/ape/api/providers.py [hunk 2]
```diff
@@ -216,29 +274,59 @@ def await_confirmations(self) -> "ReceiptAPI":
...
+            :class:`~ape.api.BlockGasAPI`
+        """
 
 
 @abstractdataclass
 class BlockConsensusAPI:
+    """
+    An abstract class representing block consensus-data,
+    such as PoW-related information regarding the block.
+    `EIP-3675` <https://eips.ethereum.org/EIPS/eip-3675>`__.
```
Comment (fubuloubu) [inline src/ape/api/providers.py:304]:
```suggestion
    See `EIP-3675` <https://eips.ethereum.org/EIPS/eip-3675>`__ for an example of potential changes.
```

File: src/ape/api/providers.py [hunk 3]
```diff
@@ -268,69 +365,184 @@ class ProviderAPI:
...
+
+        Returns:
+            int
+        """
 
     @property
     def priority_fee(self) -> int:
+        """
+        A miner tip to incentivize them
+        to include your transaction in a block.
```
Comment (fubuloubu) [inline src/ape/api/providers.py:461]:
```suggestion
        A miner tip to incentivize them to include your transaction in a block.
```

File: src/ape/api/providers.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/api/providers.py:304]: We will have to open a separate PR for any suggestion related to providers.

**Pull request closed without merging**